build fail: wasi_nn-pytorch on manylinux_2_28 (gcc 13)

### Summary

```
CMake Deprecation Warning at build/_deps/gtest-src/CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at build/_deps/gtest-src/googletest/CMakeLists.txt:56 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Downloading SIMDJSON source
-- Adding -Og to compile flag
-- Downloading SIMDJSON source -- done
-- Downloading the WASM spec test suite
-- Downloading the WASM spec test suite -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/core
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/core -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/multi-memory
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/multi-memory -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/tail-call
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/tail-call -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/extended-const
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/extended-const -- done
-- Copying test suite to /__w/WasmEdge/WasmEdge/build/test/spec/testSuites/threads
<artificial>:(.text+0x2d99): undefined reference to `testing::Message::GetString() const'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: /tmp/cckKsLVE.ltrans0.ltrans.o: in function `WasiNNTest_PyTorchBackend_Test::TestBody()':
<artificial>:(.text+0x417e): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x4534): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x46ab): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x48[23](https://github.com/WasmEdge/WasmEdge/actions/runs/9710756208/job/26802301949?pr=3421#step:8:24)): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x499b): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x615f): undefined reference to `testing::internal::EqFailure(char const*, char const*, std::string const&, std::string const&, bool)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x64ca): undefined reference to `testing::internal::EqFailure(char const*, char const*, std::string const&, std::string const&, bool)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x66ce): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x68ac): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x79[26](https://github.com/WasmEdge/WasmEdge/actions/runs/9710756208/job/26802301949?pr=3421#step:8:27)): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x79f8): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x7ac5): undefined reference to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: /tmp/cckKsLVE.ltrans0.ltrans.o:<artificial>:(.text+0x7b90): more undefined references to `testing::internal::GetBoolAssertionFailureMessage(testing::AssertionResult const&, char const*, char const*, char const*)' follow
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: /tmp/cckKsLVE.ltrans4.ltrans.o: in function `testing::AssertionResult testing::internal::CmpHelperNE<WasmEdge::Runtime::Instance::FunctionInstance*, decltype(nullptr)>(char const*, char const*, WasmEdge::Runtime::Instance::FunctionInstance* const&, decltype(nullptr) const&) [clone .constprop.0]':
<artificial>:(.text+0x4fdc): undefined reference to `testing::Message::GetString() const'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: <artificial>:(.text+0x5074): undefined reference to `testing::Message::GetString() const'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: /tmp/cckKsLVE.ltrans4.ltrans.o: in function `(anonymous namespace)::createModule(std::basic_string_view<char, std::char_traits<char> >) [clone .constprop.0]':
<artificial>:(.text+0x52a3): undefined reference to `WasmEdge::Plugin::Plugin::load(std::filesystem::path const&)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: /tmp/cckKsLVE.ltrans5.ltrans.o: in function `testing::AssertionResult testing::internal::CmpHelperEQ<int, unsigned int>(char const*, char const*, int const&, unsigned int const&) [clone .constprop.0]':
<artificial>:(.text+0x2b1): undefined reference to `testing::internal::EqFailure(char const*, char const*, std::string const&, std::string const&, bool)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: /tmp/cckKsLVE.ltrans5.ltrans.o: in function `testing::AssertionResult testing::internal::CmpHelperEQ<int, unsigned int>(char const*, char const*, int const&, unsigned int const&) [clone .constprop.1]':
<artificial>:(.text+0x[39](https://github.com/WasmEdge/WasmEdge/actions/runs/9710756208/job/26802301949?pr=3421#step:8:40)1): undefined reference to `testing::internal::EqFailure(char const*, char const*, std::string const&, std::string const&, bool)'
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: /tmp/cckKsLVE.ltrans5.ltrans.o: in function `testing::AssertionResult testing::internal::CmpHelperEQ<unsigned int, int>(char const*, char const*, unsigned int const&, int const&) [clone .constprop.0]':
<artificial>:(.text+0x472): undefined reference to `testing::internal::EqFailure(char const*, char const*, std::string const&, std::string const&, bool)'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Appendix

[CI link](https://github.com/WasmEdge/WasmEdge/actions/runs/9710756208/job/26802301949)

--- a/utils/docker/Dockerfile.manylinux_2_28-build-plugins-deps
+++ b/utils/docker/Dockerfile.manylinux_2_28-build-plugins-deps
@@ -1,11 +1,6 @@
 ARG BASE=wasmedge/wasmedge:manylinux_2_28_x86_64
 FROM ${BASE}
 
-ENV PATH /opt/rh/gcc-toolset-13/root/usr/bin${PATH:+:${PATH}}
-ENV MANPATH /opt/rh/gcc-toolset-13/root/usr/share/man${MANPATH:+:${MANPATH}}
-ENV INFOPATH /opt/rh/gcc-toolset-13/root/usr/share/info${INFOPATH:+:${INFOPATH}}
-ENV PKG_CONFIG_PATH /opt/rh/gcc-toolset-13/root/usr/lib64/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}
-
 RUN cd && (yum check-update || true) && \
   yum install -y wget unzip zlib-devel zlib-static elfutils
 

--- a/utils/docker/Dockerfile.manylinux_2_28-build-plugins-deps
+++ b/utils/docker/Dockerfile.manylinux_2_28-build-plugins-deps
@@ -19,7 +19,7 @@ COPY wasi-nn/install-pytorch.sh .
 ENV PYTORCH_VERSION "1.8.2"
 ENV PYTORCH_INSTALL_TO "/root"
 ENV Torch_DIR "/root/libtorch"
-RUN [ "/bin/bash", "install-pytorch.sh", "--disable-cxx11-abi" ]
+RUN [ "/bin/bash", "install-pytorch.sh"  ]
 
 COPY wasi-crypto/build-openssl.sh .
 ENV OpenSSL_DIR "/root/openssl-1.1.1n/openssl"

--- a/utils/docker/Dockerfile.manylinux_2_28_aarch64
+++ b/utils/docker/Dockerfile.manylinux_2_28_aarch64
@@ -7,9 +7,11 @@ MAINTAINER hydai hydai@secondstate.io
 
 ADD SHA256SUM.manylinux_2_28 /root/
 
+# See /opt/rh/gcc-toolset-13/enable
 ENV PATH /opt/rh/gcc-toolset-13/root/usr/bin${PATH:+:${PATH}}
 ENV MANPATH /opt/rh/gcc-toolset-13/root/usr/share/man${MANPATH:+:${MANPATH}}
 ENV INFOPATH /opt/rh/gcc-toolset-13/root/usr/share/info${INFOPATH:+:${INFOPATH}}
+ENV LD_LIBRARY_PATH /opt/rh/gcc-toolset-13/root/usr/lib64:/opt/rh/gcc-toolset-13/root/usr/lib:${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 ENV PKG_CONFIG_PATH /opt/rh/gcc-toolset-13/root/usr/lib64/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}
 
 RUN cd && (yum check-update || true) && yum install -y openssl-devel rpm-build cmake yum-utils && \

--- a/utils/docker/Dockerfile.manylinux_2_28_x86_64
+++ b/utils/docker/Dockerfile.manylinux_2_28_x86_64
@@ -7,9 +7,11 @@ MAINTAINER hydai hydai@secondstate.io
 
 ADD SHA256SUM.manylinux_2_28 /root/
 
+# See /opt/rh/gcc-toolset-13/enable
 ENV PATH /opt/rh/gcc-toolset-13/root/usr/bin${PATH:+:${PATH}}
 ENV MANPATH /opt/rh/gcc-toolset-13/root/usr/share/man${MANPATH:+:${MANPATH}}
 ENV INFOPATH /opt/rh/gcc-toolset-13/root/usr/share/info${INFOPATH:+:${INFOPATH}}
+ENV LD_LIBRARY_PATH /opt/rh/gcc-toolset-13/root/usr/lib64:/opt/rh/gcc-toolset-13/root/usr/lib:${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 ENV PKG_CONFIG_PATH /opt/rh/gcc-toolset-13/root/usr/lib64/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}
 
 RUN cd && (yum check-update || true) && yum install -y openssl-devel rpm-build cmake yum-utils && \

